### PR TITLE
Feature/add app description

### PR DESCRIPTION
**Pull request by NkwaTambe**
Created a high level description of the power pay app,

Explaining  how the front end will connect to the backend (which technology? Which management?…)

How the data flow in the front end is going to be manipulated or managed (where  in memory is data saved? Where is the pin saved for reusability Where is the transactionId saved…), etc.

### CODE REVIEW

File: docs/power-payment-app.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- PowerPay Users: Individuals with smartphones who can initiate and manage transactions through the app.
+- PowerPay Kiosks: Physical kiosks that facilitate top-ups and withdrawals for users with smartphones and Choronko.
+- Choronko Users: Individuals who can send and receive payments using the Choronko device without needing a smartphone.
+
+## Technologies used
+
+- ReactJS for building the user interface and implementing a Progressive Web App (PWA) approach.
+- HTML and CSS for the structure and styling of the user interface.
+- RESTful APIs for communication with the backend server.
+
```
Comment (stephane-segning) [inline docs/power-payment-app.md:20]: Apart from these, we'll also have other technologies like react libraries (like navigation, state management,...), PWA and HTTP communications, under which RESTFul APIs are done

File: docs/power-payment-app.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+### User Interface:
+
+- **Registration Process**: The registration process is a series of steps that allows users to create an account and access the payment app's features. The steps typically include:
+
+    1. **Step 1 - Personal Information**: Users enter their personal details, such as name, phone number etc. This information is necessary for account creation and verification.
+
+    2. **Step 2 - Pin Setup**: Users set up a secure pin to protect their account. The pin should meet specific criteria, such as minimum length of numbers etc.
+
+    3. **Step 3 - OTP Confirmation**: Once users have entered their information, set up a pin, they are provide with an OTP in which they now confirm their account for security measures.
```
Comment (stephane-segning) [inline docs/power-payment-app.md:31]: It was discussed, that this was the other way around. A user enters a phone number alongside a name, and the system send a OTP. Then that one is used with the PUN to ensure the user phone really exists, without exposing the combination `phoneNumber`+`PIN`.

File: docs/power-payment-app.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    4. **Step 4 - Confirmation**: Once users have entered their information, set up a pin, etc . they can proceed to complete the registration process. The "Register" button is clicked to finalize the registration.
+
+    **note:** **The above registration method is for a smartphone user.**
+    **See mockup:** ![reference image](/mockups_wireframes/Registration_process.png)
+
+
+- **Registartion process for a choronko user**:As a choronko user who wants to register, all you need to do is :
+
+1. **Step 1 - Initiating Registration**: The choronko user can initiate the registration process by calling the powerpay service and then providing  their personal information such as full name , phone number etc.
```
Comment (stephane-segning) [inline docs/power-payment-app.md:41]: He's not calling the PPService, he's sending an SMS to the API Gateway that will redirect it to our Translation service, which will send it to our PPService... but what does the Chronko user process doing here? I wanted you to discuss only the App. The choronko is not using the App.

File: docs/power-payment-app.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+## Technologies used
+
+- ReactJS for building the user interface and implementing a Progressive Web App (PWA) approach.
+- HTML and CSS for the structure and styling of the user interface.
+- RESTful APIs for communication with the backend server.
+
+## Architecture and Interaction
+
+### User Interface:
```
Comment (stephane-segning) [inline docs/power-payment-app.md:23]: Here I was expecting you to talk only about different interfaces and help yourself with wireframes. Not to explain the relation ship between those interfaces, we already have the workflow part for that.

File: docs/power-payment-app.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+3. **Step 3 -  OTP Verification**:The Choronko user will be prompted to enter the OTP or verification code received via SMS. The system will validate the code to ensure the mobile number is associated with the user and proceed to the next step upon successful verification.
+
+4. **Step 4 -  Setting a Personal Identification Number (PIN)**: For security purposes, the Choronko user will be prompted to set a PIN .The system will guide the user through the process of setting a secure PIN following specific requirements.
+
+5. **Step 5 -  Completing Registration**: Once the user has set their PIN, the registration process for the Choronko user is considered complete. The system will store the user's information and PIN securely in the backend, associating it with the user's mobile number.
+
+
+
+## User Workflow
```
Comment (stephane-segning) [inline docs/power-payment-app.md:53]: In the workflow, you can now reuse the interfaces you defined earlier to explain better.

E.g: User registration:
In this process,... (basic description of the process)
1. A user enter his data in the screen (1), 
2. Then press next to go on screen (3) where he should enter the OTP he received from our backend 
3. At last, he is being redirected to screen (?) where he...

File: docs/power-payment-app.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+**see mockup**:
+![reference image](/mockups_wireframes/User_flow_interface.png) 
+![reference image](/mockups_wireframes/Transactions.png)
+
+
+## Error Handling in the PowerPay App
+The PowerPay app strives to provide a smooth user experience by gracefully handling errors that might occur during communication with the back-end server. Here's how the app will address potential issues:
+
+Types of Errors:
```
Comment (stephane-segning) [inline docs/power-payment-app.md:82]: Awesome! You just forgot user errors. E.g: What happens if a user decides to enter a false phone number?

File: docs/power-payment-app.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+- Network Errors: These occur when the user's device loses internet connectivity or cannot establish a connection to the server.
+- Server Errors: These can happen due to issues on the back-end server, such as high load, bugs, or maintenance.
+- API Errors: Errors returned from the back-end APIs due to invalid data formats, unauthorized access attempts, or other API-specific issues.
+
+##Error Handling Strategies:
+
+- Informative Error Messages: The app will display user-friendly error messages that clearly communicate the nature of the problem.for instance:
```
Comment (stephane-segning) [inline docs/power-payment-app.md:90]: Excellent! But don't write "the app will", stay simple and keep "the app displays the error message..."

It keeps more assertive about what we're doing.

File: docs/power-payment-app.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+
+## Error Handling in the PowerPay App
+The PowerPay app strives to provide a smooth user experience by gracefully handling errors that might occur during communication with the back-end server. Here's how the app will address potential issues:
+
+Types of Errors:
+
+- Network Errors: These occur when the user's device loses internet connectivity or cannot establish a connection to the server.
+- Server Errors: These can happen due to issues on the back-end server, such as high load, bugs, or maintenance.
+- API Errors: Errors returned from the back-end APIs due to invalid data formats, unauthorized access attempts, or other API-specific issues.
```
Comment (stephane-segning) [inline docs/power-payment-app.md:86]: You should also document which APIs error are often prone to appearing

File: docs/power-payment-app.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+##Error Handling Strategies:
+
+- Informative Error Messages: The app will display user-friendly error messages that clearly communicate the nature of the problem.for instance:
+"Network Error: Please check your internet connection and try again."
+"Server Error: We're currently experiencing technical difficulties. Please try again later."
+"Invalid Input: Please enter a valid phone number." (for API errors related to user input)
+
+## Error Location:
```
Comment (stephane-segning) [inline docs/power-payment-app.md:95]: To sum up, whenever doing a RESTFull call, we should handle the error. So we should discuss the architecture of the whole

File: docs/power-payment-app.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+These error handling mechanisms will be implemented in several key locations within the user flow:
+
+- User Interface (UI) Layer: When the app encounters an error during data fetching or API calls from the UI, it will trigger the error handling logic. This could be during functionalities like:
+- Registration: Network errors might prevent successful Registration attempts.
+- Top-up: Server errors could interrupt the top-up process.
+- Sending Money: API errors might occur due to invalid recipient information.
+- Balance Check: Network errors could prevent retrieving the latest balance.
+
+## API and Backend Communication:
```
Comment (stephane-segning) [inline docs/power-payment-app.md:105]: Nice. What are the APIs we're going to be using in this project?

**Review by stephane-segning — CHANGES_REQUESTED** [review]
This is a far better job than the previous one. But you didn't discussed the architecture of the react app. You didn't discuss how we'll deploy and operate it. I provided enough comment too to guide you.

File: docs/power-payment-app.md (hunk 2 above)
Comment (NkwaTambe) [reply docs/power-payment-app.md:31]: Understood sir..

ok so here, i simple just need to interchange the flow ensuring that the user is provided with th OTP before setting his/her pin....

File: docs/power-payment-app.md (hunk 3 above)
Comment (NkwaTambe) [reply docs/power-payment-app.md:41]: Please sir @stephane-segning 
 if that's the case, it implies the entire registration process for the choronko user should be removed right? 

becuase the choronko has nothing to do with the app during the registration process.

Comment (stephane-segning) [reply docs/power-payment-app.md:41]: Not only. Everything concerning the choronko user should be kicked out

Comment (NkwaTambe) [reply docs/power-payment-app.md:41]: Undestood

**Pull request closed without merging**